zkevm-circuits/src/evm_circuit/execution/codecopy.rs

```diff
@@ -0,0 +1,476 @@
+            },
+            // Code size lookup in CODECOPY.
+            Rw::Account {
+                rw_counter: 8,
+                is_write: false,
+                account_address: callee_addr,
+                field_tag: AccountFieldTag::CodeSize,
+                value: Word::from(code.bytes.len()),
+                value_prev: Word::from(code.bytes.len()),
+            },
+            // Code hash lookup in CODECOPY.
+            Rw::Account {
+                rw_counter: 9,
+                is_write: false,
+                account_address: callee_addr,
+                field_tag: AccountFieldTag::CodeHash,
+                value: code.hash,
+                value_prev: code.hash,
+            },
+        ];
+
+        let next_memory_word_size = if size.is_zero() {
+            0
+        } else {
+            (dst_addr + size as u64 + 31) / 32
+        };
+        let gas_cost_push32 = OpcodeId::PUSH32.constant_gas_cost().as_u64();
+        let gas_cost_codecopy = OpcodeId::CODECOPY.constant_gas_cost().as_u64()
+            + calc_memory_copier_gas_cost(0, next_memory_word_size, size as u64);
+        let total_gas_cost = (3 * gas_cost_push32) + gas_cost_codecopy;
+
+        let mut steps = vec![
+            ExecStep {
+                rw_indices: vec![0],
+                execution_state: ExecutionState::PUSH,
+                rw_counter: 1,
+                program_counter: 0,
+                stack_pointer: 1024,
+                gas_left: total_gas_cost,
+                gas_cost: gas_cost_push32,
+                opcode: Some(OpcodeId::PUSH32),
+                ..Default::default()
+            },
+            ExecStep {
+                rw_indices: vec![1],
+                execution_state: ExecutionState::PUSH,
+                rw_counter: 2,
+                program_counter: 33,
+                stack_pointer: 1023,
+                gas_left: total_gas_cost.sub(gas_cost_push32),
+                gas_cost: gas_cost_push32,
+                opcode: Some(OpcodeId::PUSH32),
+                ..Default::default()
+            },
+            ExecStep {
+                rw_indices: vec![2],
+                execution_state: ExecutionState::PUSH,
+                rw_counter: 3,
+                program_counter: 66,
+                stack_pointer: 1022,
+                gas_left: total_gas_cost.sub(2 * gas_cost_push32),
+                gas_cost: gas_cost_push32,
+                opcode: Some(OpcodeId::PUSH32),
+                ..Default::default()
+            },
+            ExecStep {
+                rw_indices: vec![3, 4, 5, 6, 7, 8],
+                execution_state: ExecutionState::CODECOPY,
+                rw_counter: 4,
+                program_counter: 99,
+                stack_pointer: 1021,
+                gas_left: gas_cost_codecopy,
+                gas_cost: gas_cost_codecopy,
+                opcode: Some(OpcodeId::CODECOPY),
+                memory_size: 0,
```

Comment:
should this be the unused memory size variable from above instead of 0? If not, can we just remove it since it'll be covered by the default case?

Reply:
This should be `0` since its the memory size at the start of this opcode. (Will move it to defaults). The above `memory_size` is actually unused and I'll remove that.